### PR TITLE
Run tests locally with upstream CI images.

### PR DESCRIPTION
**Pull request by rjeffman**
This patch allows local execution of playbook tests using ustream CI
testing images. Either 'podman' or 'docker' can be used to execute the
tests.

### CODE REVIEW

File: utils/run-tests.sh [hunk 1]
```diff
@@ -0,0 +1,355 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply utils/run-tests.sh]: I modified the script to use "-eu" by default. Using "-eux" is possible, although some "colors" might be "off" (sorry, I didn't care too look at the output using "-eux").

File: tests/sanity/ignore-2.12.txt [hunk 2]
```diff
@@ -57,3 +57,4 @@ utils/galaxyfy.py shebang!skip
 utils/gen_modules_docs.sh shebang!skip
 utils/lint_check.sh shebang!skip
 utils/new_module shebang!skip
+utils/run-tests.sh shebang!skip
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply tests/sanity/ignore-2.12.txt]: Added info on how to run the script to `tests/READ.md`.

**Comment by t-woerner** [comment]
It might be good to add prameters to `podman --create`: `--dns=127.0.0.1 --dns-search=test.local --no-hosts`
Otherwise `resolv.conf` and `hosts` is copied form the host.

**Comment by rjeffman** [comment]
So, the failures occur when the `ipaserver` hostname (`ipaserver.test.local`) is defined in /etc/hosts.

When running the container with `--no-hosts`, on my machine, I'm getting KDC failures if the tests take "some time". The smallest modules, like `pwpolicy` pass, but the bigger ones, like `user`, don't.

The only workaround I found, so far, is to not have `ipaserver.test.local` set on `/etc/hosts` (or on a DNS nameserver).

**Comment by t-woerner** [comment]
Running `utils/run-tests.sh user` results in `138 skipped, 8 deselected in 1.08s`.
It would be good to be able to detect if at least a single test would be run...

**Comment by rjeffman** [comment]
Added checks for tests and test suites, and log an error if the test/test suite is not found.
If the final list is empty, stop script with an error.

**Review by t-woerner — APPROVED** [review]
LGTM

**Pull request merged**